.ci/compute_projects.py

```diff
@@ -0,0 +1,186 @@
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Computes the list of projects that need to be tested from a diff.
+
+Does some things, spits out a list of projects.
+"""
+
+from collections.abc import Set
+import pathlib
+import platform
+import sys
+
+PROJECT_DEPENDENCIES = {
+    "llvm": set(),
+    "clang": {"llvm"},
+    "bolt": {"clang", "lld", "llvm"},
+    "clang-tools-extra": {"clang", "llvm"},
+    "compiler-rt": {"clang", "lld"},
+    "libc": {"clang", "lld"},
+    "openmp": {"clang", "lld"},
+    "flang": {"llvm", "clang"},
+    "lldb": {"llvm", "clang"},
+    "libclc": {"llvm", "clang"},
+    "lld": {"llvm"},
+    "mlir": {"llvm"},
+    "polly": {"llvm"},
+}
+
+DEPENDENTS_TO_TEST = {
+    "llvm": {"bolt", "clang", "clang-tools-extra", "lld", "lldb", "mlir", "polly"},
+    "lld": {"bolt", "cross-project-tests"},
+    "clang": {"clang-tools-extra", "compiler-rt", "cross-project-tests"},
```

Comment:
It would be nice to not lose track of things were supposed to be temporarily off this list. I filed
https://github.com/llvm/llvm-project/issues/132795 for LLDB, please leave a comment like old script did.

Flang was also on this list, because it shares driver with Clang, but IIRC the understand is that flang should stop relying on Clang for its driver at some point. So I'm not asking to add flang here.